### PR TITLE
Autodocs some stuff

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

Item.dm now has some docs

## Why It's Good For The Game
Gamerdocs

### CODE REVIEW

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -50,71 +53,103 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
-	var/gas_transfer_coefficient = 1 // for leaking gas from turf to mask and vice-versa (for masks right now, but at some point, i'd like to include space helmets)
-	var/permeability_coefficient = 1 // for chemicals/diseases
-	var/siemens_coefficient = 1 // for electrical admittance/conductance (electrocution checks and shit)
-	var/slowdown = 0 // How much clothing is slowing you down. Negative values speeds you up
-	var/armour_penetration = 0 //percentage of armour effectiveness to remove
-	var/list/allowed = null //suit storage stuff.
-	var/equip_delay_self = 0 //In deciseconds, how long an item takes to equip; counts only for normal clothing slots, not pockets etc.
-	var/equip_delay_other = 20 //In deciseconds, how long an item takes to put on another person
-	var/strip_delay = 40 //In deciseconds, how long an item takes to remove from another person
+	///see setup.dm for appropriate bit flags
```
Comment (optimumtact) [inline code/game/objects/items.dm]: this is not documentation

File: code/game/objects/items.dm [hunk 2]
```diff
@@ -50,71 +53,103 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
+	var/body_parts_covered = 0
+	/// for leaking gas from turf to mask and vice-versa (for masks right now, but at some point, i'd like to include space helmets)
+	var/gas_transfer_coefficient = 1
+	/// for chemicals/diseases
```
Comment (optimumtact) [inline code/game/objects/items.dm]: ditto

File: code/game/objects/items.dm [hunk 3]
```diff
@@ -50,71 +53,103 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
+	var/permeability_coefficient = 1
+	/// for electrical admittance/conductance (electrocution checks and shit)
+	var/siemens_coefficient = 1
+	/// How much clothing is slowing you down. Negative values speeds you up
+	var/slowdown = 0
+	///percentage of armour effectiveness to remove
+	var/armour_penetration = 0
+	///suit storage stuff.
```
Comment (optimumtact) [inline code/game/objects/items.dm]: ditto

File: code/game/objects/items.dm [hunk 4]
```diff
@@ -50,71 +53,103 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
 	var/list/embedding = NONE
 
-	var/flags_cover = 0 //for flags such as GLASSESCOVERSEYES
+	///for flags such as GLASSESCOVERSEYES
+	var/flags_cover = 0
 	var/heat = 0
 	///All items with sharpness of IS_SHARP or higher will automatically get the butchering component.
 	var/sharpness = IS_BLUNT
 
+	///The type of tool, engineering or mining for example
```
Comment (optimumtact) [inline code/game/objects/items.dm]: wrong

File: code/game/objects/items.dm [hunk 5]
```diff
@@ -50,71 +53,103 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
+	var/reach = 1
 
-	//The list of slots by priority. equip_to_appropriate_slot() uses this list. Doesn't matter if a mob type doesn't have a slot.
-	var/list/slot_equipment_priority = null // for default list, see /mob/proc/equip_to_appropriate_slot()
+	///The list of slots by priority. equip_to_appropriate_slot() uses this list. Doesn't matter if a mob type doesn't have a slot. For default list, see [/mob/proc/equip_to_appropriate_slot()]
+	var/list/slot_equipment_priority = null
 
-	// Needs to be in /obj/item because corgis can wear a lot of
-	// non-clothing items
+	/// Wheher dogs can wear the item: Needs to be in /obj/item because corgis can wear a lot of non-clothing items
```
Comment (optimumtact) [inline code/game/objects/items.dm]: not if they can wear the item, but instead a reference to the datum that determines behaviour if they do.

**Comment by optimumtact** [comment]
2 out of ten apply yourself

**Comment by TiviPlus** [comment]
var/gas_transfer_coefficient  is defined for several masks but does literally nothing

File: code/game/objects/items.dm (hunk 3 above)
Comment (Rohesie) [reply code/game/objects/items.dm]: Still pending.

File: code/game/objects/items.dm [hunk 6]
```diff
@@ -165,14 +201,14 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
 	else
 		return 1
 
+///Handles attacking by blobs
 /obj/item/blob_act(obj/structure/blob/B)
 	if(B && B.loc == loc)
 		qdel(src)
-
+/// this proc says it's for initializing components, but we're initializing elements too because it's you and me against the world >:)
```
Comment (Rohesie) [inline code/game/objects/items.dm]: This proc is also deprecated.

File: code/game/objects/items.dm [hunk 7]
```diff
@@ -165,14 +201,14 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
 	else
 		return 1
 
+///Handles attacking by blobs
```
Comment (Rohesie) [inline code/game/objects/items.dm]: Children proc, leave the documenting at the base of it.

File: code/game/objects/items.dm [hunk 8]
```diff
@@ -190,14 +226,16 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
-//damagetype: The type of damage the item will inflict on the user
-//BRUTELOSS = 1
-//FIRELOSS = 2
-//TOXLOSS = 4
-//OXYLOSS = 8
-//Output a creative message and then return the damagetype done
+/**Makes cool stuff happen when you suicide with an item
+  *
+  *user: The mob that is suiciding
+  *damagetype: The type of damage the item will inflict on the user
```
Comment (spookydonut) [inline code/game/objects/items.dm]: these arent parameters

File: code/game/objects/items.dm [hunk 9]
```diff
@@ -21,25 +21,28 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
 
-	//Dimensions of the icon file used when this item is worn, eg: hats.dmi
-	//eg: 32x32 sprite, 64x64 sprite, etc.
-	//allows inhands/worn sprites to be of any size, but still centered on a mob properly
+	///Dimensions of the icon file used when this item is worn, eg: hats.dmi (32x32 sprite, 64x64 sprite, etc.). Allows inhands/worn sprites to be of any size, but still centered on a mob properly
 	var/worn_x_dimension = 32
+	///Dimensions of the icon file used when this item is worn, eg: hats.dmi (32x32 sprite, 64x64 sprite, etc.). Allows inhands/worn sprites to be of any size, but still centered on a mob properly
 	var/worn_y_dimension = 32
-	//Same as above but for inhands, uses the lefthand_ and righthand_ file vars
+	///Same as for worn_x_dimension but for inhands, uses the lefthand_ and righthand_ file vars
```
Comment (spookydonut) [inline code/game/objects/items.dm]:
```suggestion
	///Same as for [worn_x_dimension][/obj/item/var/worn_x_dimension] but for inhands, uses the lefthand_ and righthand_ file vars

```

File: code/game/objects/items.dm [hunk 10]
```diff
@@ -21,25 +21,28 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
 	var/inhand_x_dimension = 32
+	///Same as for worn_y_dimension but for inhands, uses the lefthand_ and righthand_ file vars
```
Comment (spookydonut) [inline code/game/objects/items.dm]:
```suggestion
	///Same as for [worn_y_dimension][/obj/item/var/worn_y_dimension] but for inhands, uses the lefthand_ and righthand_ file vars

```

File: code/game/objects/items.dm [hunk 11]
```diff
@@ -50,71 +53,103 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
+	var/slot_flags = 0
 	pass_flags = PASSTABLE
 	pressure_resistance = 4
 	var/obj/item/master = null
 
-	var/heat_protection = 0 //flags which determine which body parts are protected from heat. Use the HEAD, CHEST, GROIN, etc. flags. See setup.dm
-	var/cold_protection = 0 //flags which determine which body parts are protected from cold. Use the HEAD, CHEST, GROIN, etc. flags. See setup.dm
-	var/max_heat_protection_temperature //Set this variable to determine up to which temperature (IN KELVIN) the item protects against heat damage. Keep at null to disable protection. Only protects areas set by heat_protection flags
-	var/min_cold_protection_temperature //Set this variable to determine down to which temperature (IN KELVIN) the item protects against cold damage. 0 is NOT an acceptable number due to if(varname) tests!! Keep at null to disable protection. Only protects areas set by cold_protection flags
+	//flags which determine which body parts are protected from heat. Use the HEAD, CHEST, GROIN, etc. flags. See setup.dm
```
Comment (spookydonut) [inline code/game/objects/items.dm]: link to the defines using []

File: code/game/objects/items.dm [hunk 12]
```diff
@@ -50,71 +53,103 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
+	///In deciseconds, how long an item takes to put on another person
+	var/equip_delay_other = 20
+	///In deciseconds, how long an item takes to remove from another person
+	var/strip_delay = 40
+	///How long it takes to resist out of the item (cuffs and such)
 	var/breakouttime = 0
 
-	var/list/attack_verb //Used in attackby() to say how something was attacked "[x] has been [z.attack_verb] by [y] with [z]"
-	var/list/species_exception = null	// list() of species types, if a species cannot put items in a certain slot, but species type is in list, it will be able to wear that item
+	///Used in attackby() to say how something was attacked "[x] has been [z.attack_verb] by [y] with [z]"
```
Comment (spookydonut) [inline code/game/objects/items.dm]: link to the attackby

File: code/game/objects/items.dm [hunk 13]
```diff
@@ -50,71 +53,103 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
 
-	mouse_drag_pointer = MOUSE_ACTIVE_POINTER //the icon to indicate this object is being dragged
+	///the icon to indicate this object is being dragged
+	mouse_drag_pointer = MOUSE_ACTIVE_POINTER
 
+	///Does it embed and if yes, what kind of embed
 	var/list/embedding = NONE
 
-	var/flags_cover = 0 //for flags such as GLASSESCOVERSEYES
+	///for flags such as GLASSESCOVERSEYES
```
Comment (spookydonut) [inline code/game/objects/items.dm]:
```suggestion
	///for flags such as [GLASSESCOVERSEYES]

```

File: code/game/objects/items.dm [hunk 14]
```diff
@@ -50,71 +53,103 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
 
 	var/block_chance = 0
 	var/hit_reaction_chance = 0 //If you want to have something unrelated to blocking/armour piercing etc. Maybe not needed, but trying to think ahead/allow more freedom
-	var/reach = 1 //In tiles, how far this weapon can reach; 1 for adjacent, which is default
+	///In tiles, how far this weapon can reach; 1 for adjacent, which is default
+	var/reach = 1
 
-	//The list of slots by priority. equip_to_appropriate_slot() uses this list. Doesn't matter if a mob type doesn't have a slot.
-	var/list/slot_equipment_priority = null // for default list, see /mob/proc/equip_to_appropriate_slot()
+	///The list of slots by priority. equip_to_appropriate_slot() uses this list. Doesn't matter if a mob type doesn't have a slot. For default list, see [/mob/proc/equip_to_appropriate_slot()]
```
Comment (spookydonut) [inline code/game/objects/items.dm]:
```suggestion
	///The list of slots by priority. equip_to_appropriate_slot() uses this list. Doesn't matter if a mob type doesn't have a slot. For default list, see [/mob/proc/equip_to_appropriate_slot]

```

File: code/game/objects/items.dm [hunk 15]
```diff
@@ -416,21 +455,23 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
 	return
 
-// called after an item is placed in an equipment slot
-// user is mob that equipped it
-// slot uses the slot_X defines found in setup.dm
-// for items that can be placed in multiple slots
-// Initial is used to indicate whether or not this is the initial equipment (job datums etc) or just a player doing it
+/**
+  *called after an item is placed in an equipment slot
+  *user is mob that equipped it
```
Comment (spookydonut) [inline code/game/objects/items.dm]: use the proper arguments formatting

File: code/game/objects/items.dm [hunk 16]
```diff
@@ -446,16 +487,18 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
 	if(slot == ITEM_SLOT_BACKPACK || slot == ITEM_SLOT_LEGCUFFED) //these aren't true slots, so avoid granting actions there
 		return FALSE
 	return TRUE
 
-//the mob M is attempting to equip this item into the slot passed through as 'slot'. Return 1 if it can do this and 0 if it can't.
-//if this is being done by a mob other than M, it will include the mob equipper, who is trying to equip the item to mob M. equipper will be null otherwise.
-//If you are making custom procs but would like to retain partial or complete functionality of this one, include a 'return ..()' to where you want this to happen.
-//Set disable_warning to TRUE if you wish it to not give you outputs.
+/**
+  *the mob M is attempting to equip this item into the slot passed through as 'slot'. Return 1 if it can do this and 0 if it can't.
```
Comment (spookydonut) [inline code/game/objects/items.dm]: ditto

File: code/game/objects/items.dm [hunk 17]
```diff
@@ -820,23 +861,21 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
 
 	return TRUE
 
-// Called before use_tool if there is a delay, or by use_tool if there isn't.
-// Only ever used by welding tools and stacks, so it's not added on any other use_tool checks.
+/// Called before use_tool if there is a delay, or by use_tool if there isn't. Only ever used by welding tools and stacks, so it's not added on any other use_tool checks.
```
Comment (spookydonut) [inline code/game/objects/items.dm]: link to use_tool

File: code/game/objects/items.dm [hunk 18]
```diff
@@ -820,23 +861,21 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
...
-// Called before use_tool if there is a delay, or by use_tool if there isn't.
-// Only ever used by welding tools and stacks, so it's not added on any other use_tool checks.
+/// Called before use_tool if there is a delay, or by use_tool if there isn't. Only ever used by welding tools and stacks, so it's not added on any other use_tool checks.
 /obj/item/proc/tool_start_check(mob/living/user, amount=0)
 	. = tool_use_check(user, amount)
 	if(.)
 		SEND_SIGNAL(src, COMSIG_TOOL_START_USE, user)
 
-// A check called by tool_start_check once, and by use_tool on every tick of delay.
+/// A check called by tool_start_check once, and by use_tool on every tick of delay.
```
Comment (spookydonut) [inline code/game/objects/items.dm]: link tool_start_check

**Review by spookydonut — CHANGES_REQUESTED** [review]

File: code/game/objects/items.dm (hunk 11 above)
Comment (Rohesie) [reply code/game/objects/items.dm]: Still pending.

Comment (spookydonut) [inline code/game/objects/items.dm]:
```suggestion
	//flags which determine which body parts are protected from heat. [See here][HEAD]

```

**Review by Rohesie — APPROVED** [review]

**Review by SpaceManiac — APPROVED** [review]
Not perfect but it's progress.

**Pull request merged**